code/_onclick/click.dm

```diff
@@ -129,7 +129,7 @@
 
 	// Allows you to click on a box's contents, if that box is on the ground, but no deeper than that
 	if(isturf(A) || isturf(A.loc) || (A.loc && isturf(A.loc.loc)))
-		if(A.Adjacent(src)) // see adjacent.dm
+		if(A.Adjacent(src) || W.reach >= get_dist(src, A)) // see adjacent.dm

```

Comment:
A weapon that only works on the same tile already works.
Look at the logic, it's Adjacent OR having a reach more or equal to the distance between you and the target.
if you're on top of eachother, there's no distance, so a reach of 0 would be "same tile only"